cockpit_api/src/main/java/com/cockpit/api/service/jiragateway/JiraGatewayService.java

```diff
@@ -64,20 +77,18 @@ public JiraGatewayService(
     @Value("${spring.jira.jiraUrl}")
     private String jiraUrl;
 
-    @Scheduled(initialDelay = 10 * ONE_SECOND, fixedDelay = ONE_MINUTE)
-    public void updateProjects() throws UnirestException {
-        // FIXME: Create a function for the call of JIRA API
-        HttpResponse<JsonNode> response = Unirest.get(jiraUrl+"/rest/api/3/project/search")
-                .basicAuth(username, token)
-                .header("Accept", "application/json")
-                .asJson();
-        // FIXME: Create a class JiraProject to map data instead of using JSONArray (with getter setter)
-        JSONArray jiraProjects = response.getBody().getObject().getJSONArray("values");
-        for (Object jiraProject: jiraProjects){
+
+    @Scheduled(initialDelay = 10 * ONE_SECOND, fixedDelay = 10 * ONE_MINUTE)
+    public void updateProjects() throws UnirestException, JsonProcessingException {
+        HttpResponse<JsonNode> jiraProjects = getJira();
+        JSONArray jiraProjectsList = jiraProjects.getBody().getObject().getJSONArray(VALUESFIELD);
```

Comment:
Method modified to return directly a JSONArray, you cannot parse JSONArray to List<Object> directly in Java, instead you use this method of parsing

Reply:
For the method getJira(), what is being used now is asJson(), but if we replace it with asObject(JiraProjectDTO.class), i think we can erase many JSONArrays

Reply:
for example: 
   public HttpResponse<JiraProjectDTO[]> getJira() throws UnirestException {
        return Unirest.get(jiraUrl+"/rest/api/3/project/search")
                .basicAuth(username, token)
                .header(HEADERKEY, HEADERVALUE)
                .asObject(JiraProjectDTO[].class);
    }

Reply:
what exactly does this change a part from the order of parsing?

Reply:
if we decided to change the parsing order, this would take time and since we are going for the fastest/easiest solutions and since the implemented solution works, I don't see the need to change the parsing order

Reply:
For me, the DTOs are created for it, it is needed, and those Json node et JSONArray are not clean, as you said, for the _fastest/easiest solutions_, I would suggest that we merge for the demo, but this part should be refactored again for the next. I do not accept that we create the DTOs but we don't use them in the api calls.